prov/verbs/src/verbs_info.c

```diff
@@ -994,6 +994,119 @@ int vrb_get_port_space(const struct fi_info *info)
 		return RDMA_PS_TCP;
 }
 
+static int vrb_gid_is_null(const union ibv_gid *gid)
+{
+	union ibv_gid null_gid;
+	memset(&null_gid, 0, sizeof(union ibv_gid));
+	/* Default gid prefix is 0xfe80 */
+	null_gid.raw[0] = 0xfe;
+	null_gid.raw[1] = 0x80;
+
+	return !memcmp(gid, &null_gid, sizeof(union ibv_gid));
+}
+
+void vrb_set_sid(enum rdma_port_space ps,
+        uint16_t port, struct sockaddr_ib *sib)
+{
+	sib->sib_sid = htonll(((uint64_t) ps << 16) + ntohs(port));
+	sib->sib_sid_mask = htonll(RDMA_IB_IP_PS_MASK);
+
+	if (port)
+		sib->sib_sid_mask |= htonll(RDMA_IB_IP_PORT_MASK);
+}
+
+static int vrb_get_rdma_addrinfo(struct ibv_context *context,
+			     uint8_t port_num, union ibv_gid *gid,
+			     uint16_t pkey, struct rdma_addrinfo **rai)
```

Comment:
I would rename this to vrb_alloc_ib_addrinfo, and just return struct rdma_addinfo *.  Though the return value is less important to me.

Reply:
Fixed.